p2p/test/transport/transport_test.go

```diff
@@ -607,3 +613,122 @@ func TestStreamReadDeadline(t *testing.T) {
 		})
 	}
 }
+
+func TestDiscoverPeerIDFromSecurityNegotiation(t *testing.T) {
```

Comment:
@marten-seemann I mostly copied these tests from vole. Was this reasonably what you were looking for testing-wise or is this too high level and specific for what I was doing?

Reply:
Let's use the host generator pattern we use in the other tests. That way we only update the single transports table and test all the transports we support.

Reply:
Should be fixed now, this look right?